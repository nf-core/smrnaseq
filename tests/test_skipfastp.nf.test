
nextflow_pipeline {

    name "Test Workflow main.nf - test_skipfastp"
    script "main.nf"
    profile "test_skipfastp"
    tag "skipfastp"
    tag "pipeline"

    test("test_skipfastp") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 64 },

                { assert snapshot(
                    path("$outputDir/mirna_quant/mirtop/joined_samples_mirtop.tsv").exists(),
                    path("$outputDir/mirna_quant/mirtop/mirna.tsv").exists(),
                ).match("mirna_quant_mirtop") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.idxstats").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.stats").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N3_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.flagstat").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.flagstat").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.idxstats").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.stats").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N3_mature.sorted.flagstat")
                ).match("mirna_quant_bam") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_unmapped_read_counts.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_unmapped_read_counts.txt").exists()
                ).match("mirna_quant_edger_qc") },

                { assert snapshot(
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-report.html").exists(),
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-stats-contamination_basic.tsv"),
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-stats-length.tsv"),
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-stats-mirna-complexity.tsv"),
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-stats-contamination_detailed.tsv"),
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-stats-phred.tsv"),
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-stats-rnatype.tsv"),
                    path("$outputDir/mirtrace/Clone1_N1/mirtrace-stats-qcstatus.tsv"),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-report.html").exists(),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-stats-contamination_basic.tsv"),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-stats-length.tsv"),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-stats-mirna-complexity.tsv"),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-stats-contamination_detailed.tsv"),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-stats-phred.tsv"),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-stats-rnatype.tsv"),
                    path("$outputDir/mirtrace/Clone1_N3/mirtrace-stats-qcstatus.tsv")
                ).match("mirtrace") },

                { assert snapshot(
                    path("$outputDir/genome_quant/bam/Clone1_N3_mature_hairpin_genome.sorted.flagstat"),
                    path("$outputDir/genome_quant/bam/Clone1_N3_mature_hairpin_genome.sorted.idxstats"),
                    path("$outputDir/genome_quant/bam/Clone1_N1_mature_hairpin_genome.sorted.idxstats"),
                    path("$outputDir/genome_quant/bam/Clone1_N1_mature_hairpin_genome.sorted.flagstat"),
                    path("$outputDir/genome_quant/bam/Clone1_N3_mature_hairpin_genome.sorted.stats"),
                    path("$outputDir/genome_quant/bam/Clone1_N1_mature_hairpin_genome.sorted.stats")
                ).match("genome_quant_bam") },

                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_report.html").exists()
                ).match("multiqc") },

                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_data/fastqc-status-check-heatmap.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_complexity_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools-stats-dp.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_qc_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_length_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_counts_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtrace_rna_categories_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools_alignment_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtop_read_count_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtop_unique_read_count_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/mirtop_mean_read_count_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_sources.txt").exists()
                ).match("multiqc_multiqc_data") }
            )
        }
    }

}
