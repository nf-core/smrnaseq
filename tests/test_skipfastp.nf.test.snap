{
    "mirtrace": {
        "content": [
            true,
            "mirtrace-stats-contamination_basic.tsv:md5,34005177b3038da76236b5d8fb468825",
            "mirtrace-stats-length.tsv:md5,847f38c5a6bbed4d3200ecaab05e090a",
            "mirtrace-stats-mirna-complexity.tsv:md5,d175bfcbcdebbdf768217da0078ad055",
            "mirtrace-stats-contamination_detailed.tsv:md5,141f6f46a4f9fcfac91303c125a2fc75",
            "mirtrace-stats-phred.tsv:md5,b55f5d6fe7a79dafd618e13f66505dfe",
            "mirtrace-stats-rnatype.tsv:md5,424cc10a33b239d01765f981dbe4b60c",
            "mirtrace-stats-qcstatus.tsv:md5,7c41c36e544b9aafe68bbeade3de9428",
            true,
            "mirtrace-stats-contamination_basic.tsv:md5,109ca0f96520aaa21bea6f55dc1b4b03",
            "mirtrace-stats-length.tsv:md5,e37044dc40483a893b577ebdbf4fa585",
            "mirtrace-stats-mirna-complexity.tsv:md5,bd8e19d2c40c3b1dbfec4037ca288b14",
            "mirtrace-stats-contamination_detailed.tsv:md5,d116d38bc104139f7496da66ce0a9436",
            "mirtrace-stats-phred.tsv:md5,3e1b63adcc5acd334059ebbaa5141b72",
            "mirtrace-stats-rnatype.tsv:md5,8a94743f0e853f8804ae225d4a588e6f",
            "mirtrace-stats-qcstatus.tsv:md5,469f837e7403e160eaec4c1cf3af0ba3"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-11T14:28:00.975156754"
    },
    "genome_quant_bam": {
        "content": [
            "Clone1_N3_mature_hairpin_genome.sorted.flagstat:md5,22d4dffd7b6fc1d2ae7827de3fb68ad7",
            "Clone1_N3_mature_hairpin_genome.sorted.idxstats:md5,2620288b88bba1ea3315414016c083a1",
            "Clone1_N1_mature_hairpin_genome.sorted.idxstats:md5,e0e4a95f5c21a926f7894cf1fbe3110b",
            "Clone1_N1_mature_hairpin_genome.sorted.flagstat:md5,62208acf0c7418d590b41318d7e17d67",
            "Clone1_N3_mature_hairpin_genome.sorted.stats:md5,1c03f98d16d2e418692d3227c8be2803",
            "Clone1_N1_mature_hairpin_genome.sorted.stats:md5,e5aee4c4e099d5f4e82bd22f58309e2c"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T20:19:25.61285999"
    },
    "software_versions": {
        "content": [
            "{BOWTIE_MAP_GENOME={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_HAIRPIN={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_MATURE={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_SEQCLUSTER={bowtie=1.3.0, samtools=1.16.1}, CSVTK_JOIN={csvtk=0.30.0}, DATATABLE_MERGE={r-base=3.6.2}, FASTQC_RAW={fastqc=0.12.1}, FORMAT_HAIRPIN={fastx_toolkit=0.0.14}, FORMAT_MATURE={fastx_toolkit=0.0.14}, INDEX_HAIRPIN={bowtie=1.3.0}, INDEX_MATURE={bowtie=1.3.0}, MIRTOP_COUNTS={mirtop=0.4.28}, MIRTOP_EXPORT={mirtop=0.4.28}, MIRTOP_GFF={mirtop=0.4.28}, MIRTOP_STATS={mirtop=0.4.28}, MIRTRACE_QC={mirtrace=1.0.1}, PARSE_HAIRPIN={seqkit=2.6.1}, PARSE_MATURE={seqkit=2.6.1}, SAMTOOLS_FLAGSTAT={samtools=1.21}, SAMTOOLS_IDXSTATS={samtools=1.21}, SAMTOOLS_INDEX={samtools=1.21}, SAMTOOLS_SORT={samtools=1.21}, SAMTOOLS_STATS={samtools=1.21}, SEQCLUSTER_COLLAPSE={seqcluster=1.2.9}, Workflow={nf-core/smrnaseq=v2.3.2dev}}"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-02T13:45:55.267854723"
    },
    "mirna_quant_bam": {
        "content": [
            "Clone1_N3_mature_hairpin.sorted.flagstat:md5,1630edf055b591303d7c68d013745938",
            true,
            "Clone1_N3_mature_hairpin.sorted.stats:md5,b41149b9cb862c46d60ef21498f29e40",
            true,
            "Clone1_N3_mature_hairpin.sorted.idxstats:md5,b44fb26f6be2accc7d52bc38efff69f4",
            true,
            "Clone1_N3_mature.sorted.stats:md5,b1e54f83ae8a387a2071eda43ef3f9cb",
            true,
            true,
            true,
            "Clone1_N3_mature.sorted.idxstats:md5,bde0293f0938a8a074ad3ac633d8cb73",
            "Clone1_N3_mature.sorted.flagstat:md5,9e287eb7ac83624b262864d0255217fd"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-19T15:26:42.634182624"
    },
    "mirna_quant_edger_qc": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:38:26.540699935"
    },
    "multiqc_multiqc_data": {
        "content": [
            "fastqc-status-check-heatmap.txt:md5,a16737b9ae7b9b70b0ef7e462101a729",
            "fastqc_overrepresented_sequences_plot.txt:md5,3c1ffe7d55bbf2815e6bc427a2d27a2c",
            "mirtrace_complexity_plot.txt:md5,e27fb1e870985b3fe76744c027ce1c40",
            "multiqc_citations.txt:md5,f46d2983044658a4a89bdec5ba20fda3",
            "samtools-stats-dp.txt:md5,d1854b0ed73a4c9ae62a3a625c19d4b2",
            "fastqc_sequence_length_distribution_plot.txt:md5,ff2def0eab8321d4ed590b483641f43b",
            "multiqc_general_stats.txt:md5,3f5a47801a45a1b62bb519bea4896ab4",
            "fastqc_per_base_n_content_plot.txt:md5,c345fe5430e3a17ad1dbcc14e7595f50",
            "fastqc_per_sequence_quality_scores_plot.txt:md5,edf4d21e2928d37d94bb33a25e1d92a6",
            "mirtrace_qc_plot.txt:md5,49c178fc849a1aa44781ddc67c85927c",
            "mirtrace_length_plot.txt:md5,7023ffcd95379998adbd65204b9998ee",
            true,
            "fastqc_sequence_counts_plot.txt:md5,a8680d3b059401e71c7c0fe2404f5933",
            "mirtrace_rna_categories_plot.txt:md5,606174b789bdb9841f8e99206d147bb9",
            "samtools_alignment_plot.txt:md5,2778636ce5b6c9432b67014382fc35af",
            "fastqc_per_base_sequence_quality_plot.txt:md5,60f539c88c503680c0b2603749494948",
            "mirtop_read_count_plot.txt:md5,420dda9dbd3348b5c685e888dbe2a85a",
            "mirtop_unique_read_count_plot.txt:md5,ade512aac4f23e63239b7db54d2544c8",
            "mirtop_mean_read_count_plot.txt:md5,9f9cee399a861fd17f0627fc843b7c15",
            "fastqc_sequence_duplication_levels_plot.txt:md5,7e7eb4105b8f963bdf68e422e4ebce67",
            "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,7ac995de6a861676f64879b02d04f819",
            "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,c18bf431a08ec1230720d83781e8903b",
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-20T16:48:44.52005314"
    },
    "multiqc": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:38:26.610014871"
    },
    "mirna_quant_mirtop": {
        "content": [
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T20:19:25.557700049"
    }
}